custom_news_feed/custom_news_feed.py

```diff
@@ -267,11 +277,27 @@ def display_news_content(self, news_json_file_path):
             QgsMessageLog.logMessage(u'Error Reading Config file, missing field ' + str(e),'Custom News Feed')
 
 
+    def toggle_message_hashfile(self, event):
+        """Toggles pinned messages resized/full-sized state"""
+        hash = hashlib.md5(str(self.current_pinned_message["Text"]).encode('utf-8')).hexdigest()
+        if self.check_hashfile(hash) == True :
+            self.delete_hashfile(hash)
+        else:                      
+            self.create_hashfile(hash)
+        self.reloadNews
+
+
     def configure_pinned_message(self, pinnedMessageJson):
         """Adds pinned message to plugin in three possible styles."""
         self.dockwidget.pinned_message.setVisible(False)
-        self.dockwidget.pinned_message.setText(pinnedMessageJson["Text"])
-        if pinnedMessageJson["Text"] is not "":
+        self.dockwidget.pinned_message.mousePressEvent = self.toggle_message_hashfile
+        self.dockwidget.pinned_message.setText(str(self.current_pinned_message["Text"]))
+        if self.check_hashfile(hashlib.md5(str(pinnedMessageJson["Text"]).encode('utf-8')).hexdigest()) == True :
```

Comment:
PP: Das Erstellen eines MD5-Hashes in eine separate Funktion auslagern, da dies doch mehrmals im Code vorkommt. Die Funktion nimmt ein JSON entgegen und liefert den Hash zurück. So spart man sich die Wiederholungen von der String-Konvertierung, Encoding und `hexdigest`.